test/extensions/clusters/common/logical_host_integration_test.cc

```diff
@@ -10,60 +10,49 @@ namespace Extensions {
 namespace Clusters {
 
 // Logical Host integration test.
-class LogicalHostIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
```

Comment:
what happens if you let this stay a parameterized test? Was it broken in some other way besides MSAN?

Reply:
We will need to put the first_address_string_ in a shared_ptr and pass a copy to the lamda the same way as address_.  It's not a big deal. But as the race issue is not related to the IP stack, I prefer keep it as simple as possible.

Reply:
I think it's not complicated at all to keep the ip-address paramerization, and use a `std::shared_ptr<std::pair<int, absl::string_view>>` to hold the address and first address string.

I kind of want 2 methods to run run in this test to help us gain evidence we are not bleeding threads.

Cool?

Reply:
done